src/main/webapp/app/shared/notification/notification-popup/notification-popup.component.ts

```diff
@@ -40,14 +49,22 @@ export class NotificationPopupComponent implements OnInit {
      * @param notification {Notification}
      */
     navigateToTarget(notification: Notification): void {
-        this.router.navigateByUrl(this.notificationTargetRoute(notification));
+        // tslint:disable-next-line:triple-equals
```

Comment:
why disable triple-equals? why can't you use it here?

Reply:
Well, I tried all kinds of type casts, etc. but no matter what I tried I always get `false`.
Both sides have the type string and both have the exact same string (I tried fixing this issue for quite a while but with no luck sadly). It seems that enums when using  strict type correctness in typescript are only equal to the exact same enum and to nothing else (see the screenshot of my debug)
![image](https://user-images.githubusercontent.com/65814168/124452294-8c0dc180-dd86-11eb-83d7-1afe43ccdddf.png)
Do you know how to compare Enums and strings in ts correctly?

Reply:
1. there are no enums involved there, you are only using string constants, as far as I see.
2. there is a tiny little bug that unfortunately messed up your logic. The statement as it is now will _first_ add the "prefix-string" to the title and _then_ compare the resulting string value to the constant. The result of this is always false, as shown in your picture. You should add `()`: `console.log("=== without .toString() :" + (notification.title === LIVE_EXAM_EXERCISE_UPDATE_NOTIFICATION_TITLE));` the result here is: `=== without .toString() :true`; This mistake is also the reason why your prefixes get lost during logging. In summary, you could simply use `===` instead of `==` and do not have to change anything else.

Reply:
Oh yes, my mistake.